airbyte-webapp/src/pages/SettingsPage/pages/NotificationPage/components/WebHookForm.tsx

```diff
@@ -1,95 +1,84 @@
-  color: ${({ theme }) => theme.greyColor40};
-`;
-
-const FeedbackCell = styled(Cell)`
-  &:last-child {
-    text-align: left;
-  }
-  padding-left: 11px;
-`;
+const enum WebhookAction {
+  Test = "test",
+  Save = "save",
+}
 
-const Success = styled.div`
-  font-size: 13px;
-  color: ${({ theme }) => theme.successColor};
-`;
+interface FormActionType {
+  [WebhookAction.Test]: boolean;
+  [WebhookAction.Save]: boolean;
+}
 
-const Error = styled(Success)`
-  color: ${({ theme }) => theme.dangerColor};
-`;
+interface WebHookFormProps {
+  webhook: WebhookPayload;
+}
 
 const webhookValidationSchema = yup.object().shape({
   webhook: yup.string().url("form.url.error"),
   sendOnSuccess: yup.boolean(),
   sendOnFailure: yup.boolean(),
 });
 
-interface WebHookFormProps {
-  webhook: WebhookPayload;
-  successMessage?: React.ReactNode;
-  errorMessage?: React.ReactNode;
-  onSubmit: (data: WebhookPayload) => void;
-  onTest: (data: WebhookPayload) => void;
-}
-
-const WebHookForm: React.FC<WebHookFormProps> = ({ webhook, onSubmit, successMessage, errorMessage, onTest }) => {
+export const WebHookForm: React.FC<WebHookFormProps> = ({ webhook }) => {
+  const [webhookViewGuide, setWebhookViewGuide] = useState(false);
+  const [formAction, setFormAction] = useState<FormActionType>({ test: false, save: false });
+  const [errorMessage, setErrorMessage] = useState<boolean>(false);
+  const { updateWebhook, testWebhook } = useWorkspace();
   const { formatMessage } = useIntl();
 
-  const feedBackBlock = (dirty: boolean, isSubmitting: boolean, webhook?: string) => {
-    if (successMessage) {
-      return <Success>{successMessage}</Success>;
-    }
+  let timeoutID: ReturnType<typeof setTimeout> | null = null;
 
-    if (errorMessage) {
-      return <Error>{errorMessage}</Error>;
+  const webhookChange = async (action: WebhookAction, data: WebhookPayload) => {
+    setFormAction((value) => ({ ...value, [action]: true }));
+    if (action === WebhookAction.Test) {
+      await testWebhookAction(data);
     }
+    if (action === WebhookAction.Save) {
+      await updateWebhook(data);
+    }
+    setFormAction((value) => ({ ...value, [action]: false }));
+  };
 
-    if (dirty) {
-      return (
-        <Button isLoading={isSubmitting} type="submit">
-          <FormattedMessage id="form.saveChanges" />
-        </Button>
-      );
+  const testWebhookAction = async (data: WebhookPayload) => {
+    try {
+      const response = await testWebhook(data);
+      if (response.status === "failed") {
+        setErrorMessage(true);
+      }
+    } catch (e) {
+      setErrorMessage(true);
     }
 
-    if (webhook) {
-      return (
-        <Button isLoading={isSubmitting} type="submit">
-          <FormattedMessage id="settings.test" />
-        </Button>
-      );
+    if (timeoutID) {
+      clearTimeout(timeoutID);
     }
 
-    return null;
+    timeoutID = setTimeout(() => {
+      setErrorMessage(false);
+      timeoutID = null;
+    }, 2000);
```

Comment:
This custom logic should not be needed here. The message should be visible until the user presses "test" again.

Reply:
Fixed. Thanks :smile: